src/lxc/tools/lxc_top.c

```diff
@@ -631,7 +732,7 @@ int lxc_top_main(int argc, char *argv[])
 			stats_print_header(&total);
 		}
 
-		for (i = 0; i < active_cnt && i < ct_print_cnt; i++) {
+		for (i = 0; i < active_cnt && i < (ct_print_cnt || batch); i++) {
```

Comment:
Please, can you explain this change?

Reply:
You're right. I tried on my end and it only prints the container statistics of one container. Thanks for catching that @mihalicyn 

```
Name:           a2
State:          RUNNING
PID:            4916
Link:           vethq4uXbP
 TX bytes:      9.20 KiB
 RX bytes:      0 bytes
 Total bytes:   9.20 KiB
```

```
time_ms,container,cpu_nanos,cpu_sys_userhz,cpu_user_userhz,blkio_bytes,blkio_iops,mem_used_bytes,memsw_used_bytes,kernel_mem_used_bytes
1722022106544,a1,280167000,17,10,2785280,0,4681728,0,1327104
``` 

In the above example, a2 is running, but only a1 prints